### ISSUE TITLE
glue/sst2 testing dataset is problematic, so we should adjust our finetuning script

### ISSUE THREAD

**Issue by chenmoneygithub** [keras-team/keras-nlp#175]
The test split looks incorrect, here is a snapshot: 

```
((<tf.Tensor: shape=(6,), dtype=string, numpy=
array([b'uneasy mishmash of styles and genres .',
       b"this film 's relationship to actual tension is the same as what christmas-tree flocking in a spray can is to actual snow : a poor -- if durable -- imitation .",
       b'by the end of no such thing the audience , like beatrice , has a watchful affection for the monster .',
       b'director rob marshall went out gunning to make a great one .',
       b'lathan and diggs have considerable personal charm , and their screen rapport makes the old story seem new .',
       b'a well-made and often lovely depiction of the mysteries of friendship .'],
      dtype=object)>,), <tf.Tensor: shape=(6,), dtype=int32, numpy=array([-1, -1, -1, -1, -1, -1], dtype=int32)>)
```

There are two problems:
1) The train and validation split use label 0 and 1 for negative and positive reviews, while test split has -1.
2) The snapshot contains both positive and negative reviews, while all have the same label -1.

We should consider filing a bug to tensorflow dataset team, and meanwhile put up a note that our finetuning script should skip sst2 testing set (perhaps just report on validation dataset).

**Comment by mattdangerw** [comment]
This is actually by design! For sst-2 the test set is hidden (hence the -1). This is also true for most (but not all) other glue tasks. The only true way to evaluate all the GLUE scores is to prepare a submission. Essentially having private labels like this is a good way to make sure people don't "game the system" when scoring a model.

This is why we have https://github.com/keras-team/keras-nlp/issues/115
